collect_app/src/main/java/org/odk/collect/android/utilities/ViewUtil.java

```diff
@@ -0,0 +1,35 @@
+package org.odk.collect.android.utilities;
+
+
+import android.os.Build;
+import android.view.View;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ViewUtil {
+
+    private static final AtomicInteger sNextGeneratedId = new AtomicInteger(1);
+
+    public static int generateViewId() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            return View.generateViewId();
+        } else {
+            return generateViewIdPre17();
+        }
+    }
+
+    private static int generateViewIdPre17() {
+        while (true) {
```

Comment:
How often will this code result in many loops? Maybe comment why this seemingly brute-force approach is needed?